subsys/nrf_security/src/threading/threading_alt.c

```diff
@@ -11,62 +10,36 @@
 #include <string.h>
 #include <zephyr/init.h>
 #include <zephyr/kernel.h>
+#include <zephyr/sys/__assert.h>
 
-#include "nrf_security_mutexes.h"
-
-#if !defined(MBEDTLS_CONFIG_FILE)
-#include "mbedtls/config.h"
-#else
-#include MBEDTLS_CONFIG_FILE
-#endif
+K_MUTEX_DEFINE(mbedtls_threading_key_slot_mutex);
+K_MUTEX_DEFINE(mbedtls_threading_psa_globaldata_mutex);
+K_MUTEX_DEFINE(mbedtls_threading_psa_rngdata_mutex);
 
-NRF_SECURITY_MUTEX_DEFINE(mbedtls_threading_key_slot_mutex);
-NRF_SECURITY_MUTEX_DEFINE(mbedtls_threading_psa_globaldata_mutex);
-NRF_SECURITY_MUTEX_DEFINE(mbedtls_threading_psa_rngdata_mutex);
-
-static void mbedtls_mutex_init_fn(mbedtls_threading_mutex_t * mutex)
+void mbedtls_mutex_init_fn(mbedtls_threading_mutex_t *mutex)
 {
-    if(!k_is_pre_kernel() && !k_is_in_isr()) {
-        nrf_security_mutex_init(mutex);
-    }
+	__ASSERT_NO_MSG(!k_is_pre_kernel() && !k_is_in_isr());
+	k_mutex_init(mutex);
```

Comment:
I kept the old design but you are right its better to just define them without the macro. Since the PSA core does run the init function for these mutexes.

From the docs I didn't see any issues if you initialize them twice but  it is a more logical design to avoid the macros. I will update.

Reply:
I don't mean to say that one way is better over the other, but just that with how it is currently there is no need for `k_mutex_init()`.

Reply:
I didn't meant to say that there is a better either. I just though that if the PSA core initializes these mutexes it makes more sense to avoid using the macro but I think my assumption is wrong actually because nrf_security is always using this file here and not the threading.c

But k_mutex_init is still needed because the PSA core uses some mutexes which they are not allocated statically. 
More specifically this one:
https://github.com/nrfconnect/sdk-mbedtls/blob/98603a8c91660beac00e0ee1d76198fb7c4ed29b/library/memory_buffer_alloc.c#L63

Reply:
Ah hmm okay. So it might make more sense to not initialize the mutexes in the macro if `mbedtls_mutex_init_fn()` will be called with these mutexes.

Reply:
Discussed it outside of github, in short there are multiple mutexes with different initialization requirements.